src/routes/MapExplorer/MapExplorer.js

```diff
@@ -252,13 +273,66 @@ export default class MapExplorer extends React.Component {
     this.setState({ showFeatureFilters: !this.state.showFeatureFilters })
   }
 
+  toggleSavedSearches = (e) => {
+    this.setState({ showSavedSearches: !this.state.showSavedSearches })
+  }
+
+  saveSearch = (name) => {
+    const { currentBoundingBox } = this.state
+    const { currentUser, filters, location, posts } = this.props
+    const { featureTypes, search: searchText, topics } = filters
+    const { pathname } = location
+
+    let context
+    if (isAllCommunitiesPath(pathname)) context = 'all'
+    if (isPublicPath(pathname)) context = 'public'
+    if (isCommunityPath(pathname)) context = 'community'
+    if (isNetworkPath(pathname)) context = 'network'
+
+    let communitySlug
+    let networkSlug
+
+    if (context === 'community') communitySlug = getCommunitySlug(pathname)
+    if (context === 'network') networkSlug = getNetworkSlug(pathname)
+
+    const userId = currentUser.id
+
+    const postTypes = Object.keys(featureTypes).reduce((selected, type) => {
+      if (featureTypes[type]) selected.push(type)
+      return selected
+    }, [])
+
+    const lastPostId = get(posts, '[0].id')
+
+    const topicIds = topics.map(t => t.id)
+
+    const boundingBox = [
+      { lat: currentBoundingBox[1], lng: currentBoundingBox[0] },
+      { lat: currentBoundingBox[3], lng: currentBoundingBox[2] }
+    ]
+
+    const attributes = { boundingBox, communitySlug, context, lastPostId, name, networkSlug, postTypes, searchText, topicIds, userId }
+
+    this.props.saveSearch(attributes).then(_ => this.props.fetchSavedSearches())
```

Comment:
are you calling fetch to add it to the list? the reducer could also add it to the list, not a big deal but definitely more efficient than another server call

Reply:
This is related to my comment above, I'm not getting the full saved search record after creating a saved search

Reply:
@tibetsprague I'm using the reducer to add it to the list now, see this [latest commit](https://github.com/Hylozoic/hylo-evo/pull/663/commits/a740eb8d09dfd825db56e012a96fb10209c1a30f)